File: CS/MVVMExpenses/Models/Transaction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;

namespace MVVMExpenses.DataModels {
    public enum TransactionType {
        Expense,
        Income
    }
    public class Transaction {
        [Key, Display(AutoGenerateField = false)]
        public long ID { get; set; }
        [Display(AutoGenerateField = false)]
        public long AccountID { get; set; }
        [Display(Name = "ACCOUNT")]
        public virtual Account Account { get; set; }
        [Display(AutoGenerateField = false)]
        public long CategoryID { get; set; }
        [Display(Name = "CATEGORY")]
        public virtual Category Category { get; set; }
        [DataType(DataType.Date)]
        [Display(Name = "DATE")]
        public DateTime Date { get; set; }
        [DataType(DataType.Currency)]
        [Display(Name = "AMOUNT")]
        public decimal Amount { get; set; }
        [DataType(DataType.MultilineText)]
        [Display(Name = "COMMENT")]
        public string Comment { get; set; }
    }
}

File: VB/MVVMExpenses/Models/Transaction.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.ComponentModel.DataAnnotations

Namespace MVVMExpenses.DataModels
    Public Enum TransactionType
        Expense
        Income
    End Enum
    Public Class Transaction
        <Key, Display(AutoGenerateField:=False)>
        Public Property ID() As Long
        <Display(AutoGenerateField:=False)>
        Public Property AccountID() As Long
        <Display(Name:="ACCOUNT")>
        Public Overridable Property Account() As Account
        <Display(AutoGenerateField:=False)>
        Public Property CategoryID() As Long
        <Display(Name:="CATEGORY")>
        Public Overridable Property Category() As Category
        <DataType(DataType.Date), Display(Name:="DATE")>
        Public Property [Date]() As Date
        <DataType(DataType.Currency), Display(Name:="AMOUNT")>
        Public Property Amount() As Decimal
        <DataType(DataType.MultilineText), Display(Name:="COMMENT")>
        Public Property Comment() As String
    End Class
End Namespace